server/app/views/applicant/IneligibleBlockView.java

```diff
@@ -26,19 +31,66 @@ public final class IneligibleBlockView extends ApplicationBaseView {
   }
 
   public Content render(
-      Request request, String programTitle, Optional<String> applicantName, Messages messages) {
+      Request request,
+      ReadOnlyApplicantProgramService roApplicantProgramService,
+      Optional<String> applicantName,
+      Messages messages,
+      long applicantId,
+      long programId) {
```

Comment:
**suggestion (non-blocking):** instead of passing these in as params, consider adding methods to ReadOnlyApplicantProgramService for getting them

Reply:
I did this for programId, but it seems applicantId is harder to get (not stored in applicantData), so would involve a larger refactor.

Reply:
ahhh bummer